src/lm.jl

```diff
@@ -243,3 +243,46 @@ function confint(obj::LinearModel; level::Real=0.95)
     hcat(coef(obj),coef(obj)) + stderror(obj) *
     quantile(TDist(dof_residual(obj)), (1. - level)/2.) * [1. -1.]
 end
+
+"""
+    cooksdistance(obj::LinearModel)
+Compute Cook's distance for each observation, an estimate of the influence of each data point.
+Currently only model without weights or with normalised weights are supported.
+reference: https://en.wikipedia.org/wiki/Cook%27s_distance
+Credit to Tyler Beason https://github.com/tbeason
+"""
+function cooksdistance(obj::LinearModel)
+    u = residuals(obj)
+    mse = dispersion(obj,true)
+    k = dof(obj)-1
+    d_res = dof_residual(obj)
+    X = modelmatrix(obj)
+    wts = obj.rr.wts
+    if isempty(wts)
+        hii = diag(X * inv(X' * X) * X')
+    else
+        if mean(wts) == 1. # assume that the weights are normalised if the mean is 1 
+            u = @. u * sqrt(wts)
+            W = Diagonal(wts)
+            hii = diag(X * inv(X' * W * X) * X' * W)
+        else
+            error("Currently only support LinearModel without weight or with normalised weights `wts= wts ./ mean(wts)`.")
+        end
+    end
+    D = @. u^2 * (hii / (1 - hii)^2) / (k*mse)
+    return D
+end
+
+using StatsModels
+
+"""
+    cooksdistance(obj :: StatsModels.TableRegressionModel)
+wrapper from TableRegressionModel to LinearModel. Will produce error when use with something else than LinearModel.
+"""
+function cooksdistance(obj :: StatsModels.TableRegressionModel)
+    if isa(obj.model , LinearModel)
+        cooksdistance(obj.model)
+    else
+        error("cooksdistance not implemented for model type: ", typeof(obj.model))
+    end
+end
```

Comment:
Can you make a PR to add this to StatsModels instead?

Reply:
sure, can you tell me where (which file) I should put the code?
Is there is a risk that the team from StatsModels will reject it? as I suppose that both packages will need to make a release at relatively the same time.

Reply:
It's at https://github.com/JuliaStats/StatsModels.jl/blob/0d0c2bbd56b183491e8c831a92168d4265e4ea68/src/statsmodel.jl#L123.

The team is the same, so just wait until @kleinschmidt gives his opinion about it. :-)

Reply:
Ah, actually there's a third step, as the (empty) function itself should be defined in StatsBase at https://github.com/JuliaStats/StatsBase.jl/blob/master/src/statmodels.jl. So that's 3 PRs!

Reply:
@Nosferican You might be interested in this as you define `leverage` in Econometrics.jl.

Reply:
@kleinschmidt is OK, so can you file PRs in StatsBase and StatsModels?

Reply:
Ok, I will do it when everything else is green.

Reply:
ok done:

- https://github.com/JuliaStats/StatsBase.jl/pull/679
- https://github.com/JuliaStats/StatsModels.jl/pull/221